src/NServiceBus.Core/CriticalError/CriticalError.cs

```diff
@@ -49,17 +56,43 @@ public virtual void Raise(string errorMessage, Exception exception)
             Guard.AgainstNull(nameof(exception), exception);
             LogManager.GetLogger("NServiceBus").Fatal(errorMessage, exception);
 
-            if (Endpoint == null)
+            lock (endpointCriticalLock)
```

Comment:
Got it. That's because `CriticalError` is a singleton.
Wouldn't then the preference be to use concurrent collection? (just curious)